Mathlib/RingTheory/Invariant/Defs.lean

```diff
@@ -0,0 +1,30 @@
+/-
+Copyright (c) 2024 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+import Mathlib.Algebra.Algebra.Defs
+
+/-!
+# Invariant Extensions of Rings
```

Comment:
What other content will this file contain in the future?

Reply:
But it seemed natural to split off the definition at this point seeing as it now has two somewhat orthogonal use cases (one focusing on rings and prime ideals and the other just focusing on fields).